internal/provider/helpers.go

```diff
@@ -0,0 +1,256 @@
+package provider
+
+import (
+	"fmt"
+	"strings"
+
+	eboptions "github.com/coder/envbuilder/options"
+	"github.com/coder/serpent"
+	"github.com/coder/terraform-provider-envbuilder/internal/tfutil"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/spf13/pflag"
+)
+
+const (
+	envbuilderOptionPrefix = "ENVBUILDER_"
+)
+
+// nonOverrideOptions are options that cannot be overridden by extra_env.
+var nonOverrideOptions = map[string]struct{}{
+	"ENVBUILDER_CACHE_REPO": {},
+	"ENVBUILDER_GIT_URL":    {},
```

Comment:
Suggestion: While struct saves space, boolean produces nicer code and doesn't need to check "ok" on the map, simply `if overrides[name] {`.